### PR TITLE
Powershell to Convert or Split an instance (introducing Audit instances)

### PR DESCRIPTION
**Pull request by mikeminutillo**
Powershell cmdlets to
- convert an existing ServiceControl instance into an Audit instance (Invoke-ServiceControlInstanceConvert)
- split an existing ServiceControl instance into a Main and an Audit instance (Invoke-ServiceControlInstanceSplit)

Additionally, there is code preventing a user from performing the wrong type of upgrade (Upgrade, Convert, or Split) that directs them to the correct Powershell

Alos added cmdlets and upgrade code to handle RemoteInstances as we need to be able to manage them during the upgrade to 4.0 and during a split operation

- Get-ServiceControlRemotes gives you a list of all of the remote instances for a given servicecontrol instance
- Add-ServiceControlRemote lets you add one or more remote instances to a ServiceControl instance
- Remote-ServiceControlRemote lets you remove one or more remote instances from a ServiceControl instance

### CODE REVIEW

File: src/ServiceControlInstaller.Engine/Configuration/ServiceControl/SettingsList.cs [hunk 1]
```diff
@@ -65,5 +65,11 @@ public static class ServiceControlSettings
             Name = "ServiceControl/MaintenanceMode",
             SupportedFrom = new Version(1, 18, 1)
         };
+
+        public static SettingInfo RemoteInstances = new SettingInfo
+        {
+            Name = "ServiceControl/RemoteInstances",
+            SupportedFrom = new Version(1, 47, 0)
```
Comment (danielmarbach) [inline src/ServiceControlInstaller.Engine/Configuration/ServiceControl/SettingsList.cs]: Shouldn't this be 2.0?

Comment (mikeminutillo) [reply src/ServiceControlInstaller.Engine/Configuration/ServiceControl/SettingsList.cs]: Not according to https://github.com/Particular/ServiceControl/releases/tag/1.47.0

File: src/ServiceControlInstaller.Engine/Unattended/UnattendServiceControlSplitter.cs [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+                AuditQueue = source.AuditQueue,
+                AuditLogQueue = source.AuditLogQueue,
+                ForwardAuditMessages = source.ForwardAuditMessages,
+                AuditRetentionPeriod = source.AuditRetentionPeriod,
+                TransportPackage = source.TransportPackage,
+                ConnectionString = source.ConnectionString,
+                HostName = source.HostName,
+                Name = $"{source.Name}.Audit",
+                ServiceAccount = source.Service.Account,
+                // NOTE: This should always be blank
```
Comment (HEskandari) [inline src/ServiceControlInstaller.Engine/Unattended/UnattendServiceControlSplitter.cs]: hmmm what should be blank? ☝️

File: src/ServiceControlInstaller.Engine/Unattended/UnattendServiceControlToAuditInstanceConverter.cs [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+                ConnectionString = source.ConnectionString,
+                InstallPath = source.InstallPath,
+                DBPath = source.DBPath,
+                LogPath = source.LogPath,
+                HostName = source.HostName,
+                Port = source.Port,
+                DatabaseMaintenancePort = source.DatabaseMaintenancePort,
+                Name = source.Name,
+                ServiceAccount = source.Service.Account,
+                // NOTE: This should always be blank
```
Comment (HEskandari) [inline src/ServiceControlInstaller.Engine/Unattended/UnattendServiceControlToAuditInstanceConverter.cs]: Sam as above

File: src/ServiceControlInstaller.Engine/Unattended/UnattendServiceControlSplitter.cs (hunk 2 above)
Comment (mikeminutillo) [reply src/ServiceControlInstaller.Engine/Unattended/UnattendServiceControlSplitter.cs]: The password. We never read it from the instance so it should always be blank

**Pull request merged**